src/autorun.ts

```diff
@@ -53,25 +86,40 @@ export const run = <T>(fn: Cb<T>): Observable<T> => new Observable(observer => {
         return false;
     }
 
+    const removeUnusedDeps = (ofType: TrackType) => {
+        for (let [key, { used, subscription, trackType }] of deps.entries()) {
+            if (used || trackType > ofType) {
+                continue;
+            }
+            subscription?.unsubscribe();
+            deps.delete(key);
+        }
+    }
+
     const runFn = () => {
         if (hasError) {
             return throwError(error);
         }
-        const prev$ = context.$;
-        const prev_ = context._;
-        context.$ = $;
-        context._ = _;
+        // Mark all deps as untracked and unused
+        for (let dep of deps.values()) {
+            dep.track = false;
+            dep.used = false;
+        }
+        const prevCtxt = {...context};
+        Object.assign(context, {$, _});
```

Comment:
**optional:**

uhm... I think I initiated this verbose code 🙂
and if we change `context` to be declared as `let`, we could simply:

```js
const prevCtxt = context;
context = {$, _};
```

Would that be enough?

Reply:
Yea the idea was to make it a little less verbose, and I thought it was done like this because the mutability of content of the context object needed to be exposed to the outside world. But I now realize that is not the case, and we could as well make the `context` reference itself mutable instead of the content... Mutability is hard to reason about sometimes! I'll fix.